src/Sandbox.Core/Database.cs

```diff
@@ -45,9 +47,18 @@ public async Task RemoveHousemates(IEnumerable<Housemate> housemates)
         await connection.ExecuteAsync(sql, housemates);
     }
 
+    public Task<bool> LogData<K>(IList<K> data) where K : IData
```

Comment:
```suggestion
    public Task<bool> LogData<TData>(IList<TData> data) where TData : IData
```
Standard practice is for a generic to be `T`, but if that is already used by another generic then use a description of the generic type prefixed with `T`. You can see examples of this in .net lib code https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.priorityqueue-2?view=net-8.0